### DISCUSSION TITLE
Thoughts on adding a ULID struct?

### DISCUSSION THREAD

**Discussion by pdevito3** [dotnet/csharplang#7495] [Language Ideas] [answered]
I came across [this article](https://shopify.engineering/building-resilient-payment-systems) by shopify and thought point 6 was interesting about [ULID](https://github.com/ulid/spec)s.

The short version is they were able to realize a 50 percent decrease in INSERT statement duration by switching from UUIDv4 to ULID for idempotency keys.

Now, most of the operations in the majority of applications are reads vs writes, but this seemed interesting and figured i'd float the idea. I think the closest I've seen in our ecosystem at the moment is [Cysharp/Ulid](https://github.com/Cysharp/Ulid). There's also [Newid](https://github.com/phatboyg/NewId) but that's a bit different.

**Comment by alexrp** [comment, marked as answer]
This seems like a discussion/proposal for [dotnet/runtime](https://github.com/dotnet/runtime). The C# language would need no changes for this.

**Comment by 333fred** [comment]
This sounds like a runtime api request, so I would suggest asking in the dotnet/runtime repo.
